src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.NativeAot.cs

```diff
@@ -507,6 +524,26 @@ public object GetOrRegisterObjectForComInstance(IntPtr externalComObject, Create
                         retValue = handle.Target;
                         return true;
                     }
+
+                    if (wrapperMaybe is not null)
+                    {
+                        retValue = wrapperMaybe;
+                        NativeObjectWrapper wrapper = new NativeObjectWrapper(
+                            externalComObject,
+                            this,
+                            retValue);
+                        if (!_rcwTable.TryGetValue(retValue, out var existingWrapper))
+                        {
+                            _rcwTable.Add(retValue, wrapper);
+                        }
+                        else
+                        {
+                            throw new NotSupportedException();
```

Comment:
`NativeObjectWrapper` finalizer is still going to run when this exception is thrown. Does it match the CoreCLR behavior?